src/OpenTelemetry/Logs/ILogger/OpenTelemetryLoggerProvider.cs

```diff
@@ -96,6 +96,9 @@ void ISupportExternalScope.SetScopeProvider(IExternalScopeProvider scopeProvider
     /// <inheritdoc/>
     public ILogger CreateLogger(string categoryName)
     {
+        // Lock-free reading leveraging Hashtable's thread safety feature.
```

Comment:
This comment does not fit here. May need to explain reasoning behind it.

Reply:
this accidently got part of this pr from my local change... However, the comment is valid - it is explaining why we are not locking when reading from the hashtable, but locking when writing to it...

Elsewhere (like in metrics), we use `ConcurrentDictionary`, so the choice of `HashTable` here is to take advantage of its special thread safety guarantee.